cmd/setup-type.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+// SetupType - enum for setup type.
+type SetupType int
+
+const (
+	// FSSetupType - FS setup type enum.
+	FSSetupType SetupType = iota + 1
+
+	// XLSetupType - XL setup type enum.
+	XLSetupType
+
+	// DistXLSetupType - Distributed XL setup type enum.
+	DistXLSetupType
+)
+
```

Comment:
@balamurugana looks like the below two functions String() and IsValid() are not used anywhere?